test/runtests.jl

```diff
@@ -253,6 +253,14 @@ end
 @test sprint(JSON.print, Float64) == string("\"Float64\"")
 @test_throws ArgumentError sprint(JSON.print, JSON)
 
+# test for issue #90 - Date/DateTime
+if VERSION >= v"0.4.6-pre+18"
```

Comment:
Check for 0.4 or `isdefined(Base, :Dates)` or conditionally require/import `Dates` should be good enough.